include/rwqueue.h

```diff
@@ -26,30 +26,44 @@
 #include <condition_variable>
 #include <deque>
 #include <mutex>
+#include <vector>
 
 template <typename T>
 class RWQueue {
```

Comment:
How about adding some class level comment about this intended use of this queue? Most importantly, I think it should be mentioned that it's thread-safe and it was written with concurrent async publishers & consumers in mind.

Reply:
added; great suggestion.